webhook/main.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2025 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"html"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"slices"
+	"strconv"
+	"syscall"
+	"time"
+
+	cloudbuild "cloud.google.com/go/cloudbuild/apiv1/v2"
+	"cloud.google.com/go/cloudbuild/apiv1/v2/cloudbuildpb"
+	kms "cloud.google.com/go/kms/apiv1"
+	"github.com/abcxyz/pkg/githubauth"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/sethvargo/go-gcpkms/pkg/gcpkms"
+	"golang.org/x/oauth2"
+
+	"github.com/google/go-github/v69/github"
+	"github.com/googleapis/gax-go/v2"
+)
+
+// gcloud run deploy webhook-go --region=us-west1 --source . --update-secrets=${WEBHOOK_KEY_PATH}=${KEY_NAME}:latest --allow-unauthenticated --set-env-vars=APP_ID=${APP_ID},TRIGGER_ID=${TRIGGER_ID},PROJECT_ID=${PROJECT_ID},KEY_ID=${KEY_ID},TRIGGER_NAME=${TRIGGER_NAME},LOCATION=${LOCATION},WEBHOOK_KEY_PATH=${WEBHOOK_KEY_PATH}
+
+type Server struct {
+	logger           *slog.Logger
+	webhookSecret    []byte
+	appClient        *githubauth.App
+	cloudBuildClient CloudBuildClient
+	baseURL          *url.URL
+}
+
+type CloudBuildClient interface {
+	RunBuildTrigger(ctx context.Context, req *cloudbuildpb.RunBuildTriggerRequest, opts ...gax.CallOption) (*cloudbuild.RunBuildTriggerOperation, error)
+}
+
+//nolint:unused // main is the entry point of the program and is called by Cloud Run.
+func main() {
+	ctx, done := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	defer done()
+
+	logger := logging.NewFromEnv("WEBHOOK_RECEIVER_")
+	ctx = logging.WithLogger(ctx, logger)
+
+	if err := realMain(ctx, logger); err != nil {
+		done()
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+}
+
+//nolint:unused
+func realMain(ctx context.Context, logger *slog.Logger) error {
+	webhookSecret, err := getWebhookSecret(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	kmsClient, err := getKeyManagementClient(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	signer, err := getSigner(ctx, logger, kmsClient)
+	if err != nil {
+		return err
+	}
+
+	appClient, err := getAppClient(ctx, logger, signer)
+	if err != nil {
+		return err
+	}
+
+	cloudBuildClient, err := getCloudBuildClient(ctx, logger)
+	if err != nil {
+		return err
+	}
+
+	server := &Server{
+		logger:           logger,
+		webhookSecret:    webhookSecret,
+		appClient:        appClient,
+		cloudBuildClient: cloudBuildClient,
+	}
```

Comment:
nit; this isn't very idiomatic go. Typically you'd combine all of the `getXXX` methods into a single method called something like `NewServer`. That would create all of the clients, read the secrets, whatever and then return a *Server. 

The collection of calls above that boils down to 

```
x, err := getX()
if err != nil {
    return nil
}
```

and is an anti pattern. If you are going to do it this way with separate get methods for each thing then you should be wrapping every one of those err objects not just returning it up the stack.